orchagent/mirrororch.cpp

```diff
@@ -320,91 +316,128 @@ bool MirrorOrch::getNeighborInfo(const string& name, MirrorEntry& session)
 {
     SWSS_LOG_ENTER();
 
-    NeighborEntry neighbor;
-    MacAddress mac;
-
-    assert(session.nexthopInfo.resolved);
-
-    SWSS_LOG_INFO("Getting neighbor info for %s session\n", name.c_str());
-
-    if (!m_neighOrch->getNeighborEntry(session.nexthopInfo.nexthop, neighbor, mac))
+    // 1) Check if the destination IP is directly connected and get neighbor
+    // 2) Check if the session has next hop and neighbor information is available
+    if (!m_neighOrch->getNeighborEntry(session.dstIp,
+                session.neighborInfo.neighbor, session.neighborInfo.mac) &&
+            (session.nexthopInfo.nexthop == IpAddress() ||
```

Comment:
>session.nexthopInfo.nexthop == IpAddress()  [](start = 13, length = 43)

is there a function saying check nexthop.empty() so that it is more clear what you are checking here?

Reply:
yes. there is a function `isZero`. I will use that.